.github/workflows/test.yaml

```diff
@@ -6,29 +6,44 @@ on:
       - main
 
 jobs:
-  test:
-    strategy:
-      matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
-      fail-fast: false
-
-    runs-on: ${{ matrix.os }}
+  static:
```

Comment:
Moved static checks to only run once on Ubuntu. It could be argued that `clippy` should run everywhere but we don't have any platform specific code in Kitsune itself yet. Happy to move it back if that changes or if people just think that's more correct